src/lib/processSnapshot.ts

```diff
@@ -16,28 +16,294 @@ export default class Queue {
     private processing: boolean = false;
     private processingSnapshot: string = '';
     private ctx: Context;
+    private snapshotNames: Array<string> = [];
+    private variants: Array<string> = [];
 
     constructor(ctx: Context) {
         this.ctx = ctx;
     }
 
-    enqueue(item: Snapshot): void {
+    enqueue(item: Snapshot, start: boolean): void {
         this.snapshots.push(item);
+        if(start){
+            if (!this.processing) {
+                this.processing = true;
+                this.processNext();
+            }
+        }
+    }
+
+    startProcessingfunc(): void {
         if (!this.processing) {
             this.processing = true;
             this.processNext();
         }
     }
 
+    private processGenerateVariants(snapshot: Snapshot): void {
+        if (snapshot.options) {
+            if (snapshot.options.web) {
+                this.generateWebVariants(snapshot, snapshot.options.web);
+            }
+            if (snapshot.options.mobile) {
+                this.generateMobileVariants(snapshot, snapshot.options.mobile);
+            }
+        } 
+        
+        if (!snapshot.options || (snapshot.options && !snapshot.options.web && !snapshot.options.mobile)) {
+            this.generateVariants(snapshot, this.ctx.config);
+        }
+    }
+    
+
+    private generateVariants(snapshot: Snapshot, config: any): void {
+        // Process web configurations if they exist
+        
+        if (config.web) {
+            const browsers = config.web.browsers || [];
+            const viewports = config.web.viewports || [];
+            
+            for (const browser of browsers) {
+                for (const viewport of viewports) {
+                    const width = viewport.width;
+                    const height = viewport.height || 0;  // Use 0 if height is not provided
+                    const variant = `${snapshot.name}_${browser}_viewport[${width}]_viewport[${height}]`;
+                    this.variants.push(variant);
+                }
+            }
+        }
+    
+        // Process mobile configurations if they exist
+        if (config.mobile) {
+            const devices = config.mobile.devices || [];
+            const orientation = config.mobile.orientation || "portrait";  // Default to portrait if not provided
+            const fullPage = config.mobile.fullPage ?? true; // FullPage defaults to true if not defined
+            
+            for (const device of devices) {
+                const variant = `${snapshot.name}_${device}_${orientation}_${fullPage ? 'fullPage' : 'noFullPage'}`;
```

Comment:
Please use static constants for browsers `'fullPage'`  `'noFullPage'` values, its used 4 times in same file